### PR TITLE
Adding in ldap results to agent query

### PR DESCRIPTION
**Pull request by carolyncole**
refs #1041

### CODE REVIEW

File: app/controllers/agents_controller.rb [hunk 1]
```diff
@@ -30,9 +32,62 @@ def solr_search_results(query)
...
+      main_results.sort { |x, y| hash_key(y) <=> hash_key(x) }
+    end
+
+    def hash_key(result)
+      display_name = result['display_name']
+      display_name = display_name.first if display_name.is_a?(Array)
+      display_name = display_name.downcase if display_name.present?
+      email = result['email']
+      email = email.first if email.is_a?(Array)
+      email = email.downcase if email.present?
```
Comment (awead) [inline app/controllers/agents_controller.rb]: You're doing the same thing with email and display name. Can you put that into a method?

File: app/controllers/agents_controller.rb [hunk 2]
```diff
@@ -30,9 +32,62 @@ def solr_search_results(query)
...
+    #   "sur_name"=>["COLE"],
+    #   "psu_id"=>["cam156"],
+    #   "email"=>["cam156@psu.edu"]}]
+    def ldap_search_results(query)
+      name = Namae::Name.parse(query.upcase)
+
+      email_items = LdapDisambiguate::LdapUser.query_ldap_by_mail(query + '@psu.edu', ldap_attrs)
+      name_items = LdapDisambiguate::LdapUser.query_ldap_by_name(name.given, "#{name.family}*", ldap_attrs)
+      email_items = [] if email_items.blank?
+      name_items = [] if name_items.blank?
```
Comment (awead) [inline app/controllers/agents_controller.rb]: email_items and name_items are treated identically. How about a method that wraps the response from LdapDisambiguate::LdapUser.query_ldap_by_mail and returns [] if the response is blank?

File: spec/controllers/agents_controller_spec.rb [hunk 3]
```diff
@@ -10,6 +10,56 @@
...
+
+    let(:jamie_l_franks_ldap_entry) do
+      Net::LDAP::Entry.new('uid=qjf1,dc=psu,dc=edu').tap do |e|
+        e[:uid] = ['qjf1']
+        e[:givenname] = ['JAMIE L']
+        e[:sn] = 'FRANKS'
+        e[:mail] = ['qjf1@psu.edu']
+        e[:displayname] = ['JAMIE L FRANKS']
+      end
+    end
```
Comment (awead) [inline spec/controllers/agents_controller_spec.rb]: Can we use some kind of factory here?

Comment (carolyncole) [reply spec/controllers/agents_controller_spec.rb]: @awead good call.  There was already a factory

Comment (awead) [reply spec/controllers/agents_controller_spec.rb]: @cam156 oh nice, I didn't know there was one there.

**Pull request merged**